src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2620,6 +2557,72 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             }
 
             return true;
+
+
+        }
+
+        private static Stream GetAuxPeStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider metadataPEStreamProvider) 
+        {
+            if (metadataDiagnostics.HasAnyErrors())
+            {
+                return null;
+            }
```

Comment:
Why check for errors here? Shouldn't `SerializePeToStream` return early rather than creating the stream if there are errors? Same comment for `GetPeStream` below.